cfg/config.go

```diff
@@ -76,17 +85,34 @@ func GetConfig() (Options, error) {
 		return Options{}, err
 	}
 
-	return Options{
-		Server{
+	opt := Options{
+		LogLevel: os.Getenv(LogLvlConfigKey),
+		Server: Server{
 			Host: os.Getenv("SERV_HOST"),
 			Port: os.Getenv("SERV_PORT"),
 		},
-		DB{
+		DB: DB{
 			Host:     os.Getenv("DB_HOST"),
 			Port:     os.Getenv("DB_PORT"),
 			Username: os.Getenv("DB_USER"),
 			Password: os.Getenv("DB_PASSWORD"),
 			DBName:   os.Getenv("DB_NAME"),
 		},
-	}, nil
+	}
+
+	if err := opt.validate(); err != nil {
+		return Options{}, fmt.Errorf("validation failed: %w", err)
+	}
+
+	return opt, nil
+}
+
+func (opt *Options) validate() error {
```

Comment:
Do you really need a pointer receiver here?

Reply:
oops It's my fault. It's not needed here

Reply:
I will try to explain. In the previous version of the code, we simply added the user to the database and at the same time logged all his fields, but the ID was empty because the ID is created in the database. And it seemed normal. Now we have decided that almost all user fields are sensitive data except ID (which is empty). There is no point in logging an empty value. So I changed the code so that the user ID is returned from the database after adding